### PR TITLE
ci: add sle-micro provisioning

### PR DESCRIPTION
**Pull request by yangchiu**
ci: add sle-micro provisioning

For https://github.com/longhorn/longhorn/issues/5675

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: test_framework/Jenkinsfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ def BUILD_TRIGGER_BY = "\n${currentBuild.getBuildCauses()[0].shortDescription}"
 def SELINUX_MODE = params.SELINUX_MODE ? "SELINUX_MODE=${params.SELINUX_MODE}" : ""
 
 def CREDS_ID = JOB_BASE_NAME == "longhorn-tests-regression" ? "AWS_CREDS_RANCHER_QA" : "AWS_CREDS"
+def REGISTRATION_CODE_ID = params.ARCH == "amd64" ? "REGISTRATION_CODE" : "REGISTRATION_CODE_ARM64"
```
Comment (innobead) [inline test_framework/Jenkinsfile:9]: What reason we need to difference the value based on arch?

**Review by innobead — APPROVED** [review]
LGTM

**Comment by innobead** [comment]
Need to backport to 1.4.x and 1.3.x.

**Comment by innobead** [comment]
@Mergifyio backport v1.4.x v1.3.x

File: test_framework/Jenkinsfile (hunk 1 above)
Comment (yangchiu) [reply test_framework/Jenkinsfile:9]: The code is generated per OS version and arch in SUSE customer center. If we upgrade to sle-micro 5.4 in the future, new codes need to be generated for sle-micro 5.4 too.

**Pull request merged**